src/containerapp/azext_containerapp/custom.py

```diff
@@ -1240,6 +1244,73 @@ def list_certificates(cmd, name, resource_group_name, location=None, certificate
     return list_certificates_logic(cmd, name, resource_group_name, location, certificate, thumbprint, managed_certificates_only=managed_certificates_only, private_key_certificates_only=private_key_certificates_only)
 
 
+def upload_certificate(cmd, name, resource_group_name, certificate_file=None, certificate_name=None, certificate_password=None, location=None, prompt=False, certificate_identity = None, certificate_key_vault_url=None):
+    _validate_subscription_registered(cmd, CONTAINER_APPS_RP)
+
+    # validate arguments
+    if certificate_file and certificate_key_vault_url:
+        raise ValidationError("Cannot use certificate file/password with certificate identity/keyvaulturl at the same time")
+    if (not certificate_file) and (not certificate_key_vault_url):
+        raise ValidationError("Either certificate file/password or certificate identity/keyvaulturl should be set when hostName is set")
+
+    # construct payload
+    thumbprint = ""
```

Comment:
`thumbprint` is used to `generate_randomized_cert_name`, if the thumbprint is '', the cert name with look like this:
```
>>> generate_randomized_cert_name('','1')
'1-rg--6848'
```

Reply:
![image](https://github.com/Azure/azure-cli-extensions/assets/46143499/322e4d5f-9838-4527-8ee8-8371cd1b3000)